File: backend/src/main/java/com/badminton/booking/service/PaymentService.java
package com.badminton.booking.service;

import com.badminton.booking.dto.booking.DepositPaymentRequest;
import com.badminton.booking.dto.booking.BookingResponse;

import java.util.Map;

public interface PaymentService {
    
    /**
     * Thanh toán deposit cho booking
     */
    BookingResponse payDeposit(DepositPaymentRequest request);
    
    /**
     * Thanh toán phần còn lại khi check-in
     */
    BookingResponse payRemaining(Integer bookingId, String paymentMethod);

    /**
     * Tạo VNPay payment URL cho thanh toán deposit.
     */
    String createVnPayDepositUrl(DepositPaymentRequest request, String ipAddress);

    /**
     * Tạo VNPay payment URL cho thanh toán phần còn lại (check-in).
     */
    String createVnPayRemainingUrl(Integer bookingId, String ipAddress);

    /**
     * Xác nhận thanh toán VNPay sau khi redirect về hệ thống.
     */
    BookingResponse confirmVnPayPayment(Map<String, String> vnpParams);
}

File: frontend/src/services/paymentService.js
import axiosClient from '../axiosConfig/axiosConfig';

const paymentService = {
  // Thanh toán deposit (1/3 tiền cọc)
  payDeposit: async (depositData) => {
    try {
      const response = await axiosClient.post('/payments/deposit', depositData);
      return response.data;
    } catch (error) {
      console.error('Error paying deposit:', error);
      throw error;
    }
  },

  // Tạo VNPay payment URL cho thanh toán deposit
  createVnPayDepositUrl: async (depositData) => {
    try {
      const response = await axiosClient.post('/payments/deposit/vnpay-url', depositData);
      return response.data;
    } catch (error) {
      console.error('Error creating VNPay payment URL:', error);
      throw error;
    }
  },

  // Xác nhận thanh toán VNPay sau khi redirect về frontend
  confirmVnPayPayment: async (vnpParams) => {
    try {
      const response = await axiosClient.post('/payments/vnpay/confirm', vnpParams);
      return response.data;
    } catch (error) {
      console.error('Error confirming VNPay payment:', error);
      throw error;
    }
  },

  // Tạo VNPay payment URL cho phần còn lại khi check-in
  createVnPayRemainingUrl: async (bookingId) => {
    try {
      const response = await axiosClient.post(`/payments/remaining/${bookingId}/vnpay-url`);
      return response.data;
    } catch (error) {
      console.error('Error creating VNPay remaining payment URL:', error);
      throw error;
    }
  },

  // Thanh toán phần còn lại khi check-in (non-VNPay methods)
  payRemaining: async (bookingId, paymentMethod) => {
    try {
      const response = await axiosClient.post(`/payments/remaining/${bookingId}`, null, {
        params: { paymentMethod }
      });
      return response.data;
    } catch (error) {
      console.error('Error paying remaining:', error);
      throw error;
    }
  }
};

export default paymentService;
